homeassistant/components/prometheus/__init__.py

```diff
@@ -311,28 +313,43 @@ def _handle_lock(self, state):
         value = self.state_as_number(state)
         metric.labels(**self._labels(state)).set(value)
 
-    def _handle_climate(self, state):
-        temp = state.attributes.get(ATTR_TEMPERATURE)
+    def _handle_climate_temp(self, state, attr, metric_name, metric_description):
+        temp = state.attributes.get(attr)
         if temp:
             if self._climate_units == TEMP_FAHRENHEIT:
                 temp = fahrenheit_to_celsius(temp)
             metric = self._metric(
-                "temperature_c",
+                metric_name,
                 self.prometheus_cli.Gauge,
-                "Temperature in degrees Celsius",
+                metric_description,
             )
             metric.labels(**self._labels(state)).set(temp)
 
-        current_temp = state.attributes.get(ATTR_CURRENT_TEMPERATURE)
-        if current_temp:
-            if self._climate_units == TEMP_FAHRENHEIT:
-                current_temp = fahrenheit_to_celsius(current_temp)
-            metric = self._metric(
-                "current_temperature_c",
-                self.prometheus_cli.Gauge,
-                "Current Temperature in degrees Celsius",
-            )
-            metric.labels(**self._labels(state)).set(current_temp)
+    def _handle_climate(self, state):
+        self._handle_climate_temp(
+            state,
+            ATTR_CURRENT_TEMPERATURE,
+            "current_temperature_c",
+            "Current Temperature in degrees Celsius",
+        )
+        self._handle_climate_temp(
+            state,
+            ATTR_TEMPERATURE,
+            "temperature_c",
+            "Target temperature in degrees Celsius",
+        )
+        self._handle_climate_temp(
+            state,
+            ATTR_TARGET_TEMP_HIGH,
+            "temperature_high_c",
```

Comment:
Any objections to calling these `climate_target_temperature_high_c` and `climate_target_temperature_low_c`?

Existing metric names (especially `temperature_c`) are ambiguous and don't align with the [naming guidelines](https://github.com/home-assistant/core/blob/dev/homeassistant/components/prometheus/README.md), but renaming them would be a breaking change. New metrics, however, should have more explicit names.

Reply:
Sure, np

Reply:
Or would it be better to use labels for this, e.g. `climate_temperature{type="high|low|...}`? then if you just graph `climate_temperature` you get all of them

Reply:
Yeah, labels could work as well here. I would suggest these two new measurements to use `point="high"` and `point="low"`, while the existing target temperature should keep being reported with no additional labels set (this should be easy to verify with a test).

Note that I have just sent [a PR](https://github.com/home-assistant/core/pull/50156) renaming the existing climate temperature metrics to align with the naming guidelines. It might be easier for you to build on top of that change, but I won't mind rebasing if you get this PR finished first.

Thanks!

Reply:
There's been some pushback in that PR from one of the hass maintainers, so I suspect it might stall or be rejected eventually. I am not thrilled about adding more labels to a metric with a super ambiguous name (temperature_c), so perhaps the best way forward here would be to add two new properly-named metrics (climate_target_temperature_high_celsius and climate_target_temperature_low_celsius). WDYT?

Reply:
FYI, #50156 has finally been merged. @mdz, would you like to rebase this?

Reply:
@knyar PTAL